### PR TITLE
Update contributing guide and PR template

### PR DESCRIPTION
**Pull request by mhuggins**
## Description
This updates some contributor documentation to add clarity around chores that previously didn't fall into any category.

## Motivation and Context
There was no documented way to categorize small chores like code cleanup.

## How Has This Been Tested?
N/A

## Screenshots (if appropriate):
N/A

## Types of changes
- [x] None of these. 😄
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] I have read the **CONTRIBUTING** document.

## Closing issues
N/A

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ There is no need to squash your work as we will do that when the maintainer merg
   ```
   type(scope):desc
   ```
-  **Type** can be `fix`, `feat`, `docs`, `test` or `refactor`.
+  **Type** can be `chore`, `feat`, `docs`, or `test`.
```
Comment (jeroenptrs) [inline .github/CONTRIBUTING.md]: Could we keep `test` part of `chore`? Not sure how we can separate feature and breaking changes better than adding `breaking change`, or should we keep it in `feat`?

Comment (jeroenptrs) [reply .github/CONTRIBUTING.md]: @DalderupMaurice what do you think

Comment (DalderupMaurice) [inline .github/CONTRIBUTING.md]: Do we need to add breakingChange or such? @jeroenptrs

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -21,8 +21,8 @@
 
 ## Types of changes
 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
-- [ ] Bug fix (non-breaking change which fixes an issue)
-- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Chore (non-breaking change that fixes an issue)
```
Comment (DalderupMaurice) [inline .github/PULL_REQUEST_TEMPLATE.md]: Chore (refactor or non-breaking change that fixes an issue) maybe? :P

Comment (DalderupMaurice) [reply .github/PULL_REQUEST_TEMPLATE.md]: or Chore (non-breaking change that fixes an issue or refactor of existing code)

Comment (jeroenptrs) [reply .github/PULL_REQUEST_TEMPLATE.md]: tests, refactors and fixes

Comment (jeroenptrs) [reply .github/PULL_REQUEST_TEMPLATE.md]: I don't think mentioning non-breaking changes is necessary to be honest 🤷‍♀️

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (DalderupMaurice) [reply .github/CONTRIBUTING.md]: In my opinion test should be included in all upcoming PR's after we've decided on the muon thing, so actual PR's with "test" would be very few and thus we should put it in chore indeed 😄

Comment (DalderupMaurice) [reply .github/CONTRIBUTING.md]: feat can be both breaking or non breaking - they can just add the X of breaking change into the PR template

Comment (jeroenptrs) [reply .github/CONTRIBUTING.md]: Fine by me 👍

**Pull request merged**